src/hotspot/share/services/threadService.hpp

```diff
@@ -487,10 +487,12 @@ class JavaThreadInObjectWaitState : public JavaThreadStatusChanger {
   bool _active;
 
  public:
-  JavaThreadInObjectWaitState(JavaThread *java_thread, bool timed) :
+  JavaThreadInObjectWaitState(JavaThread *java_thread, bool timed, bool interruptible) :
     JavaThreadStatusChanger(java_thread,
-                            timed ? JavaThreadStatus::IN_OBJECT_WAIT_TIMED : JavaThreadStatus::IN_OBJECT_WAIT) {
-    if (is_alive()) {
+                            // This helper should do nothing when interruptible == false, so we set active = false.
```

Comment:
What is `active`??

Reply:
`active` is `_active`. I've removed this line of comment. It is already a good clarification in your suggestion above.
Setting `_active` to `false` makes it to overpass the calls to:
 `_stat->monitor_wait_begin()` and `_stat->monitor_wait_begin()`.